soh/soh/Enhancements/TimeSavers/SkipCutscene/Story/SkipBlueWarp.cpp

```diff
@@ -150,35 +37,143 @@ void EnKo_MoveWhenReady(EnKo* enKo, PlayState* play) {
+                       gSaveContext.cutsceneIndex == 0xFFF3) {
+                // Normally set in the blue warp cutscene
+                Flags_SetEventChkInf(EVENTCHKINF_DEATH_MOUNTAIN_ERUPTED);
+
+                gSaveContext.entranceIndex = ENTR_DEATH_MOUNTAIN_CRATER_FIRE_TEMPLE_BLUE_WARP;
+                isBlueWarpCutscene = true;
+                // Water Temple Blue warp
+            } else if (gSaveContext.entranceIndex == ENTR_CHAMBER_OF_THE_SAGES_0 && gSaveContext.cutsceneIndex == 0x0 &&
+                       gSaveContext.chamberCutsceneNum == CHAMBER_CS_WATER) {
+                // Normally set in the blue warp cutscene
+                gSaveContext.dayTime = gSaveContext.skyboxTime = 0x4800;
+                Flags_SetEventChkInf(EVENTCHKINF_RAISED_LAKE_HYLIA_WATER);
+
+                gSaveContext.entranceIndex = ENTR_LAKE_HYLIA_WATER_TEMPLE_BLUE_WARP;
+                isBlueWarpCutscene = true;
+                // Spirit Temple Blue warp
+            } else if (gSaveContext.entranceIndex == ENTR_CHAMBER_OF_THE_SAGES_0 && gSaveContext.cutsceneIndex == 0x0 &&
+                       gSaveContext.chamberCutsceneNum == CHAMBER_CS_SPIRIT) {
+                gSaveContext.entranceIndex = ENTR_DESERT_COLOSSUS_SPIRIT_TEMPLE_BLUE_WARP;
+                isBlueWarpCutscene = true;
+                // Shadow Temple Blue warp
+            } else if (gSaveContext.entranceIndex == ENTR_CHAMBER_OF_THE_SAGES_0 && gSaveContext.cutsceneIndex == 0x0 &&
+                       gSaveContext.chamberCutsceneNum == CHAMBER_CS_SHADOW) {
+                gSaveContext.entranceIndex = ENTR_GRAVEYARD_SHADOW_TEMPLE_BLUE_WARP;
+                isBlueWarpCutscene = true;
+            }
+
+            if (isBlueWarpCutscene) {
+                if (gSaveContext.entranceIndex != ENTR_LAKE_HYLIA_WATER_TEMPLE_BLUE_WARP) {
+                    // Normally set in the blue warp cutscene
+                    gSaveContext.dayTime = gSaveContext.skyboxTime = 0x8000;
+                }
+
+                *should = false;
+                gSaveContext.cutsceneIndex = 0;
+            }
+
+            // This is outside the above condition because we want to handle both first and following visits to the blue
+            // warp
+            if (sEnteredBlueWarp && overrideBlueWarpDestinations) {
+                Entrance_OverrideBlueWarp();
+            }
+        }
+
+        sEnteredBlueWarp = false;
+    });
+
+    /**
+     * Using this hook to simply observe that Link has entered a bluewarp
+     * This way we know to allow entrance rando overrides to be processed on the next tranisition hook
+     */
+    REGISTER_VB_SHOULD(VB_PLAY_BLUE_WARP_CS, { sEnteredBlueWarp = true; });
 }
+
+static RegisterShipInitFunc initFunc(RegisterSkipBlueWarp,
+                                     { CVAR_ENHANCEMENT("TimeSavers.SkipCutscene.Story"), "IS_RANDO" });
+static RegisterShipInitFunc initFunc2(RegisterShouldPlayBlueWarp);
```

Comment:
The names don't really matter but I'd prefer they be more specific rather than having `initFunc` and `initFunc2`.

Also see my other comment about the list of strings. This one might be a little hairier though and I don't know if there's a CVar or a string similar to "IS_RANDO" for boss rush...

Reply:
There is not